src/lib/format/l20n/ast/parser.js

```diff
@@ -5,25 +5,32 @@ import { L10nError } from '../../../errors';
 
 const MAX_PLACEABLES = 100;
 
-export default {
-  parse: function(env, string, pos = false) {
-    this._source = string;
-    this._index = 0;
-    this._length = string.length;
-    this._curEntryStart = 0;
+
+class ParseContext {
+  constructor(env, pos) {
     this._config = {
       pos: pos
```

Comment:
since it's not a runtime, I think I prefer to keep it as a collection and let it grow. But that's an API preference, not based on any strong reasons.

@Pike : any preference here?